doped/utils/parsing.py

```
@@ -552,23 +573,117 @@
     idx = []
-    for i, site in enumerate(structure):
+    for i, site in enumerate(structure_or_sites):
         if _parse_site_species_str(site, wout_charge=True) == species_name:
             coords.append(site.frac_coords if frac_coords else site.coords)
             idx.append(i)
 
     return np.array(coords), np.array(idx)
 
 
+def get_matching_site(
+    site: PeriodicSite | np.ndarray[float], structure: Structure, anonymous: bool = False, tol: float = 0.5
```

Pylint (3.3.7)

[error] Value 'np.ndarray' is unsubscriptable

(E1136)